src/tike/ptycho/solvers/divided.py

```diff
@@ -307,24 +356,48 @@ def _update_nearplane(op, comm, nearplane, psi, scan_, probe, unique_probe,
 
         # Update each direction
         if recover_psi:
-            weighted_step_psi[0] = comm.pool.reduce_mean(
-                weighted_step_psi,
-                axis=-5,
-            )[..., 0, 0, 0]
-            common_grad_psi[0] = comm.pool.reduce_gpu(common_grad_psi)
+            if comm.use_mpi:
+                weighted_step_psi[0] = comm.Allreduce_mean(
+                    weighted_step_psi,
+                    axis=-5,
+                )[..., 0, 0, 0]
+                common_grad_psi[0] = comm.Allreduce_reduce(
+                    common_grad_psi,
+                    dest='gpu',
+                )
+            else:
+                weighted_step_psi[0] = comm.pool.reduce_mean(
+                    weighted_step_psi,
+                    axis=-5,
+                )[..., 0, 0, 0]
+                common_grad_psi[0] = comm.reduce(common_grad_psi, 'gpu')
 
+            #test = cp.ones(common_grad_psi[0].shape, common_grad_psi[0].dtype)
```

Comment:
```suggestion
```

Reply:
@xiaodong-yu, is there a reason you want to keep these commented-out line of code?

Reply:
I overlooked this suggestion.